From 4f5a1c2c7409dee9a8e81f75141a4fd9dd889af7 Mon Sep 17 00:00:00 2001
From: Nick Cao <nickcao@nichi.co>
Date: Sun, 15 Jan 2023 20:15:55 +0800
Subject: [PATCH] tpm2_context_init: fix driver name checking

"libtss2-tcti-" is patched into "/nix/store/xxxxxxxxx-tpm2-tss-3.2.0/lib/libtss2-tcti-",
resulting in the concated filename being an absolute path, tripping
checks in filename_is_valid
---
 src/shared/tpm2-util.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/shared/tpm2-util.c b/src/shared/tpm2-util.c
index ba8dfb041d..7de5d5fc77 100644
--- a/src/shared/tpm2-util.c
+++ b/src/shared/tpm2-util.c
@@ -192,7 +192,7 @@ int tpm2_context_init(const char *device, struct tpm2_context *ret) {
                 fn = strjoina("libtss2-tcti-", driver, ".so.0");
 
                 /* Better safe than sorry, let's refuse strings that cannot possibly be valid driver early, before going to disk. */
-                if (!filename_is_valid(fn))
+                if (!path_is_valid(fn))
                         return log_error_errno(SYNTHETIC_ERRNO(EINVAL), "TPM2 driver name '%s' not valid, refusing.", driver);
 
                 dl = dlopen(fn, RTLD_NOW);
-- 
2.39.0

